index.js

```diff
@@ -13,7 +13,7 @@
  * @private
  */
 
-var matchHtmlRegExp = /["'&<>]/
+var matchHtmlRegExp = /["'&<>]/g
```

Comment:
The regex is global, so it won't start from index 0 every time if you don't reset the `lastIndex`.

Reply:
It resets on the last iteration of the while loop.

Reply:
Indeed it does. I misread the `while` as an `if`...